cleanlab_studio/utils/data_enrichment/enrich.py

```diff
@@ -33,112 +32,125 @@ def enrich_data(
         ]
     ]
 
 
-def get_regex_matches(
+def process_regex(
     column_data: Union[pd.Series, List[str]],
-    regex: Union[str, re.Pattern[str], List[re.Pattern[str]]],
-    disable_warnings: bool = False,
+    regex: Union[str, Replacement, List[Replacement]],
 ) -> Union[pd.Series, List[str]]:
     """
-    Extracts the first match to the provided regular expression pattern from each example in the provided column of data.
+    Performs regex matches or replacements to the given string according to the given matching patterns and replacement strings.
+
+    Use this function for: tuning regex replacements to obtain the best outputs from the raw LLM responses for your dataset obtained via ``enrich_data()``, without having to re-run the LLM.
 
-    Use this function for: tuning regex patterns to extract the best outputs from the raw LLM responses for your dataset obtained via ``enrich_data()``, without having to re-run the LLM.
-    If a list of regular expressions is provided, the expressions are applied in order, and the first valid regex match is returned.
+    If a string value is passed in, a regex match will be performed and the matched pattern will be returned (if the pattern cannot be matched, None will be returned).
+    Specifically the provided string will be passed into Python's `re.match()` method.
+    Pass in a tuple `(R1, R2)` instead if you wish to perform find and replace operations rather than matching/extraction.
+    `R1` should be a string containing the regex pattern to match, and `R2` should be a string to replace matches with.
+    Pass in a list of tuples instead if you wish to apply multiple replacements. Replacements will be applied in the order they appear in the list.
+    Note that you cannot pass in a list of strings (chaining of multiple regex processing steps is only allowed for replacement operations).
 
-    **Note:** Regex patterns should each specify exactly 1 group that is represents the desired characters to be extracted from the raw response using parenthesis like so ``'(<desired match group pattern>)'``.
-    **Example 1:** `r'.*The answer is: (Bird|[Rr]abbit).*'` will extract strings that are the words 'Bird', 'Rabbit' or 'rabbit' after the characters "The answer is: " from the raw response text.
-    **Example 2:** `r'.*(\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b).*'` will match an email in the raw response LLM response.
+    **Example 1:** ``regex = '.*The answer is: (Bird|[Rr]abbit).*'`` will extract strings that are the words 'Bird', 'Rabbit' or 'rabbit' after the characters "The answer is: " from the raw response.
+    **Example 2:** ``regex = [('True', 'T'), ('False', 'F')]`` will replace the words True and False with T and F.
+    **Example 3:** ``regex = (' Explanation:.*', '') will remove everything after and including the words "Explanation:".
+    For instance, the response "True. Explanation: 3+4=7, and 7 is an odd number." would return "True" after the regex replacement.
```

Comment:
Wouldn't this return "True."

Reply:
> **Example 3:** ``regex = (' Explanation:.*', '') will remove everything after and including the words "Explanation:".
>     For instance, the response "True. Explanation: 3+4=7, and 7 is an odd number." would return "True" after the regex replacement.

This is stated here? Not sure if something was misunderstood

Reply:
Oh nvm I realized it was missing the period.

Reply:
Yep I think that regex would still include "."